cql3/statements/alter_keyspace_statement.cc

```diff
@@ -81,33 +86,80 @@ void cql3::statements::alter_keyspace_statement::validate(query_processor& qp, c
 #endif
 }
 
+static logging::logger mylogger("alter_keyspace");
+
 future<std::tuple<::shared_ptr<cql_transport::event::schema_change>, std::vector<mutation>, cql3::cql_warnings_vec>>
 cql3::statements::alter_keyspace_statement::prepare_schema_mutations(query_processor& qp, const query_options&, api::timestamp_type ts) const {
+    using namespace cql_transport;
     try {
-        auto old_ksm = qp.db().find_keyspace(_name).metadata();
+        event::schema_change::target_type target_type = event::schema_change::target_type::KEYSPACE;
+        auto ks = qp.db().find_keyspace(_name);
+        auto ks_md = ks.metadata();
+
         const auto& tm = *qp.proxy().get_token_metadata_ptr();
         const auto& feat = qp.proxy().features();
+        auto ks_md_update = _attrs->as_ks_metadata_update(ks_md, tm, feat);
+        std::vector<mutation> muts;
+        std::vector<sstring> warnings;
+
+        if (ks.get_replication_strategy().uses_tablets() && !rf_map.empty()) {
+            qp.db().real_database().validate_keyspace_update(*ks_md_update);
+            if (auto repl_strategy = _attrs->get_replication_strategy_class();
+                    repl_strategy.has_value() && repl_strategy != "NetworkTopologyStrategy")
```

Comment:
nit: missing curly braces around throw statement

Reply:
also, I can't see how repl_strategy can be disengaged, but if it is we should probably throw this error as well,
i.e. `repl_strategy.value_or("") != "NetworkTopologyStrategy"` or
`!repl_strategy || *repl_strategy != "NetworkTopologyStrategy"`

Reply:
> also, I can't see how repl_strategy can be disengaged

We don't have to provide repl strategy when ALTERing, e.g. this statement doesn't provide it:
```
ALTER KEYSPACE xxx WITH durable_writes=true;
```
so `repl_strategy` doesn't have to be set (that's why it's `std::optional`), and we shouldn't throw an error if someone skips it.